src/openstudio_lib/ResultsTabView.cpp

```diff
@@ -360,7 +359,7 @@ void ResultsView::comboBoxChanged(int index) {
 
   m_progressBar->setError(false);
 
-  QUrl url(filename);
+  QUrl url = QUrl::fromLocalFile(filename);
```

Comment:
because we use this.

https://doc.qt.io/qt-6/qurl.html#fromLocalFile

[qDebug](https://doc.qt.io/qt-6/qtlogging.html#qDebug)() << [QUrl](https://doc.qt.io/qt-6/qurl.html#QUrl)::fromLocalFile("/home/user/file.txt"); // QUrl("file:///home/user/file.txt")

Reply:
Does `QUrl::fromLocalFile` escape bad characters (like `#`) in the path?  I don't see that in the documentation for the method.  I'll see if I can add a unit test for that.

Reply:
It seems like it does indeed, since I no longer get the "file#1.osm" isssue.

I checked the source code to make sure, basically it sets the Scheme to fileScheme, and eventually the q_urlRecode is called:

https://github.com/qt/qtbase/blob/175569e1ec7f4d2ff7ace442c23c0bcdb86063d0/src/corelib/io/qurlrecode.cpp#L603-L639

Reply:
Chain of call: 

https://github.com/qt/qtbase/blob/175569e1ec7f4d2ff7ace442c23c0bcdb86063d0/src/corelib/io/qurl.cpp#L3345-L3346

https://github.com/qt/qtbase/blob/175569e1ec7f4d2ff7ace442c23c0bcdb86063d0/src/corelib/io/qurl.cpp#L1111-L1115

https://github.com/qt/qtbase/blob/175569e1ec7f4d2ff7ace442c23c0bcdb86063d0/src/corelib/io/qurl.cpp#L808

Reply:
And the characters handled are these ones: https://github.com/qt/qtbase/blob/175569e1ec7f4d2ff7ace442c23c0bcdb86063d0/src/corelib/io/qurl.cpp#L727-L738